File: src/main/java/ar/edu/unq/epers/unidad4/service/impl/ItemServiceImpl.java
package ar.edu.unq.epers.unidad4.service.impl;

import ar.edu.unq.epers.unidad4.model.Item;
import ar.edu.unq.epers.unidad4.persistence.sql.ItemDAO;
import ar.edu.unq.epers.unidad4.service.interfaces.ItemService;
import jakarta.persistence.EntityNotFoundException;
import org.springframework.stereotype.Service;

import java.util.Collection;

@Service
public class ItemServiceImpl implements ItemService {

    private final ItemDAO itemDAO;

    public ItemServiceImpl(ItemDAO itemDAO) {
        this.itemDAO = itemDAO;
    }

    @Override
    public Item guardar(Item item) {
        return itemDAO.save(item);
    }

    @Override
    public Item recuperar(Long itemId) {
        return itemDAO.findById(itemId).orElseThrow(() -> new EntityNotFoundException("No se encontro un item con id " + itemId));
    }

    @Override
    public Collection<Item> getMasPesados(int peso) {
        return itemDAO.getMasPesados(peso);
    }

    @Override
    public Collection<Item> getItemsDePersonajesDebiles(int vida) {
        return itemDAO.getItemsDePersonajesDebiles(vida);
    }

    @Override
    public void clearAll() {
        itemDAO.deleteAll();
    }
}

File: src/main/kotlin/ar/edu/unq/epers/unidad4/service/ItemServiceImpl.kt
package ar.edu.unq.epers.unidad4.service

import ar.edu.unq.epers.unidad4.dao.ItemDAO
import ar.edu.unq.epers.unidad4.exception.EntityNotFoundException
import ar.edu.unq.epers.unidad4.model.Item
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ItemServiceImpl : ItemService {

    @Autowired lateinit var itemDAO: ItemDAO

    override fun guardar(item: Item): Item {
        return itemDAO.save(item)
    }

    override fun recuperar(itemId: Long): Item {
        return itemDAO.findById(itemId).orElseThrow { EntityNotFoundException("Invalid item #$itemId") }
    }

    override fun getMasPesados(peso: Int): Collection<Item> {
        return itemDAO.getMasPesados(peso)
    }

    override fun getItemsPersonajesDebiles(vida: Int): Collection<Item> {
        return itemDAO.getItemsPersonajesDebiles(vida)
    }

    override fun clearAll() {
        itemDAO.detachDelete()
    }
}